server/app/middlewares/websocket_backend.rb

```diff
@@ -9,8 +9,10 @@ class WebsocketBackend
   WATCHDOG_THRESHOLD = 1.0.seconds
   WATCHDOG_TIMEOUT = 60.0.seconds
 
+  STRFTIME = '%F %T.%NZ'
   KEEPALIVE_TIME = 30.seconds
   PING_TIMEOUT = Kernel::Float(ENV['WEBSOCKET_TIMEOUT'] || 5.seconds)
+  CLOCK_SKEW = Kernel::Float(ENV['KONTENA_CLOCK_SKEW'] || 1.seconds)
```

Comment:
The acceptable clock skew into the future can be very small (1s), but the acceptable clock skew into the past is `CLOCK_SKEW + PING_TIMEOUT`, because the websocket open will always have some network delay.